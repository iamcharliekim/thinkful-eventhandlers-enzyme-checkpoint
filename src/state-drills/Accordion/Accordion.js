import React from 'react'

class Accordion extends React.Component {

	state = {
		sectionIndex: ''
	}
	
	onClickHandler = (i) => {
		this.setState({ sectionIndex: i})
	}
	
	
	render() {
		let output = <li></li>
		
		if (this.props.sections){
			output = this.props.sections.map((section, i)=>{
				return ( 
						<li key={i}>
							<button onClick={()=> this.onClickHandler(i)}> 	
								{section.title}
							</button>

							{this.state.sectionIndex !== '' && this.state.sectionIndex === i ? <p>{this.props.sections[this.state.sectionIndex].content}</p> : null }
						</li>
				)
			})
			
		}

		return <ul>{output}</ul>
	}
}

export default Accordion