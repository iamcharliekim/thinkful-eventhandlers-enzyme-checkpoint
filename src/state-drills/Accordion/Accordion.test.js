import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

// The component renders an empty li when the sections prop is not supplied
it('the Accordion component renders an empty li when the sections prop is not supplied', ()=>{
	const tree = renderer.create(<Accordion/>).toJSON();
	expect(tree).toMatchSnapshot()
})

// The component renders no sections as active by default
it('the Accordion component renders no sections as active by default', ()=>{
	const tree = renderer.create(<Accordion sections={sections}/>).toJSON();
	expect(tree).toMatchSnapshot()
})

// The component opens a clicked section
it('the Accordion component opens a clicked section', ()=>{
	const wrapper = shallow(<Accordion sections={sections}/>)
	
	wrapper.find('button').at(0).simulate('click')
	expect(toJson(wrapper)).toMatchSnapshot()
})

// The component only opens the last section when multiple sections have been clicked. 

it('the Accordion component only opens 1 section at a time', ()=>{
	const wrapper = shallow(<Accordion sections={sections}/>)
	wrapper.find('button').at(0).simulate('click')
	wrapper.find('button').at(1).simulate('click')
	
	expect(toJson(wrapper)).toMatchSnapshot()
})
