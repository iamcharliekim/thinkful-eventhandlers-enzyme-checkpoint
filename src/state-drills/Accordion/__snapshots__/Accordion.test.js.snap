// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Accordion component only opens 1 section at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`the Accordion component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`the Accordion component renders an empty li when the sections prop is not supplied 1`] = `
<ul>
  <li />
</ul>
`;

exports[`the Accordion component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
